src/backend/renderer/damage.rs

```diff
@@ -360,14 +320,27 @@ impl OutputDamageTracker {
     /// next call to [`render_output`](OutputDamageTracker::render_output)
     pub fn from_output(output: &Output) -> Self {
         Self {
-            mode: OutputDamageTrackerMode::Auto(output.clone()),
+            mode: OutputModeSource::Auto(output.clone()),
             last_state: Default::default(),
             span: info_span!("renderer_damage", output = output.name()),
         }
     }
 
+    /// Initialize a new [`OutputDamageTracker`] from an [`OutputModeSource`].
+    ///
+    /// This should only be used when trying to support both static and automatic output mode
+    /// sources. For known modes use [`OutputDamageTracker::new`] or
+    /// [`OutputDamageTracker::from_output`] instead.
+    pub fn from_mode_source(output_mode_source: OutputModeSource) -> Self {
+        Self {
+            mode: output_mode_source,
+            last_state: Default::default(),
+            span: info_span!("render_damage"),
+        }
+    }
+
```

Comment:
Same about these constructors. This is not part of the released api and even that I wouldn't care too much about stability here. Lets just do one constructor for `impl Into<OutputModeSource>` and implement that for `Output` and a static tuple (or introduce another small completely public struct for static size, scale and transform).

Reply:
I don't think that really makes things easier. We already have "a small tuple", it's the `OutputModeSource` variant.

It's just going to get incredibly complicated and difficult for people to figure out which types are allowed here by relying on an overwhelming amount of from/into impls.

Another method isn't that difficult, but it can make reading the API much, much simpler.